File: Python/Algorithms/922.py
class Solution:
    def sortArrayByParityII_inplace(self, A):
        for idx, i in enumerate(A):
            type_ = idx % 2
            if not type_ == i % 2:
                for t in range(idx + 1, len(A)):
                    if (A[t] % 2) == type_:
                        A[idx], A[t] = A[t], A[idx]
                        break
        return A

    def sortArrayByParityII(self, A):
        # build a list of the same size.
        res = [0 for _ in A]
        oddp, evenp = 1, 0
        # add according to oddp(osition) and evenp(osition)
        for i in A:
            if i % 2:
                res[oddp] = i
                oddp += 2
            else:
                res[evenp] = i
                evenp += 2
        return res

File: Rust/Algorithms/922.rs
impl Solution {
    pub fn sort_array_by_parity_ii(a: Vec<i32>) -> Vec<i32> {
        let mut res = vec![0; a.len()];
        let [mut oddp, mut evenp] = [1, 0];
        for i in a {
            if i % 2 == 1 {
                res[oddp] = i;
                oddp += 2;
            } else {
                res[evenp] = i;
                evenp += 2;
            }
        }
        res
    }
}
